homeassistant/components/ecobee/binary_sensor.py

```diff
@@ -1,37 +1,42 @@
 """Support for Ecobee binary sensors."""
-from homeassistant.components import ecobee
-from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice,
+    DEVICE_CLASS_OCCUPANCY,
+)
 
-ECOBEE_CONFIG_FILE = "ecobee.conf"
+from .const import DOMAIN
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Ecobee sensors."""
-    if discovery_info is None:
-        return
-    data = ecobee.NETWORK
+    """Old way of setting up ecobee binary sensors."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up ecobee binary (occupancy) sensors."""
+    data = hass.data[DOMAIN]
     dev = list()
     for index in range(len(data.ecobee.thermostats)):
         for sensor in data.ecobee.get_remote_sensors(index):
```

Comment:
Is this returning cached results or doing I/O?

Reply:
Returning cached results. The update() method of the data object does I/O and the response is cached  in data.ecobee.thermostats. get_remote_sensors() parses the data.ecobee.thermostats property but does not do I/O.